packages/apollo-language-server/src/languageProvider.ts

```diff
@@ -351,26 +351,31 @@ ${argumentNode.description ? argumentNode.description : ""}
     _context: ReferenceContext,
     _token: CancellationToken
   ): Promise<Location[] | null> {
-    const project = this.workspace.projectForFile(uri);
-    if (!project) return null;
-    const document = project.documentAt(uri, position);
+    const projectFindReferencesRunIn = this.workspace.projectForFile(uri);
+    if (!projectFindReferencesRunIn) return null;
+    const document = projectFindReferencesRunIn.documentAt(uri, position);
     if (!(document && document.ast)) return null;
 
-    if (!project.schema) return null;
+    function getRefsInSingleProject(project: GraphQLProject) {
+      if (!project.schema) return null;
+      if (!(document && document.ast)) return null;
```

Comment:
I don't think you have to repeat this every iteration, since it's already checked above and'll be the same `document`.

Reply:
The duplicate check is required by Typescript because `document.ast` could have been mutated in between the time that the first check occurs and when this function gets called. I did hoist out `positionInDocument`.